app/models/server/raw/LivechatVisitors.ts

```diff
@@ -92,4 +93,40 @@ export class LivechatVisitorsRaw extends BaseRaw<ILivechatVisitor> {
 
 		return this.find(query, options);
 	}
+
+	findOneGuestByEmailAddress(emailAddress: string): Promise<ILivechatVisitor|null> {
+		const query = {
+			'visitorEmails.address': new RegExp(`^${ escapeRegExp(emailAddress) }$`, 'i'),
+		};
+
+		return this.findOne(query);
+	}
+
+	async getNextVisitorUsername(): Promise<string> {
+		const query = {
+			_id: 'Livechat_guest_count',
+		};
+
+		const update = {
+			$inc: {
+				value: 1,
+			},
+		};
+
+		const livechatCount = await Settings.findAndModify(query, update);
+		if (livechatCount?.value?.value) {
+			const count = livechatCount.value.value;
+			return `guest-${ Number(count) + 1 }`;
+		}
+		return Promise.resolve('');
```

Comment:
After deciding that we will use existing visitor framework instead of new visitors, I remove some code but some code missed was missed out. This is one such function. I will remove this.